core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java

```diff
@@ -804,7 +805,8 @@ public static boolean canAssignFrom(
     }
 
     if (areCharacterSetsMismatched(toType, fromType)) {
-      return false;
+      return toType.getCharset() == StandardCharsets.ISO_8859_1
+          && fromType.getCharset() == StandardCharsets.UTF_16LE;
```

Comment:
we may try to avoid this change by making changes in joinbuilder, while creating RelDataType

Reply:
Removed